src/components/Board.tsx

```diff
@@ -1,33 +1,119 @@
+import { MatchInfo, GameSpec, Piece } from '../types';
+import CanvasImage from './CanvasImage';
+import BoardPiece from './BoardPiece';
+import { AppBar, FlatButton } from 'material-ui';
 
 interface Props {
   pieces: Piece[];
+  matchInfo: MatchInfo;
+  gameSpec: GameSpec;
+  myUserId: string;
+  onReset: () => void;
 }
 
+// for testing purposes
+const testProps: Props = {
+  pieces: [],
+  matchInfo: {
+    matchId: '123',
+    gameSpecId: 'test',
+    game: {
+      gameSpecId: 'test',
+      gameName: 'Chess',
+      screenShot: {
+        imageId: '',
+        width: 100,
+        height: 100,
+        isBoardImage: false,
+        downloadURL:
+          'https://firebasestorage.googleapis.com/v0/b/universalgamemaker.appspot.com/o/images' +
+          '%2F-KtRrhplWWiDZXECJ_FO.jpg?alt=media&token=6808e2bb-cb5c-48ec-ae15-4a5d3e9a2b29'
+      }
+    },
+    participantsUserIds: ['test user'],
+    lastUpdatedOn: 0,
+    matchState: []
+  },
+  gameSpec: {
+    board: {
+      imageId: '',
+      width: 100,
+      height: 100,
+      isBoardImage: true,
+      downloadURL:
+        'https://firebasestorage.googleapis.com/v0/b/universalgamemaker.appspot.com/o/images' +
+        '%2F-KtRrhplWWiDZXECJ_FO.jpg?alt=media&token=6808e2bb-cb5c-48ec-ae15-4a5d3e9a2b29'
+    },
+    gameSpecId: 'test',
+    pieces: []
+  },
+  myUserId: 'test user',
+  onReset: () => {
+    console.log('Reset Pressed');
+  }
+};
+
 /**
  * A reusable board class, that given a board image and pieces in props
  * can draw the board and piece on top of it using konva.
  * Should also add drag and drop functionality later on.
  */
-class Board extends React.Component<Props, {}> {
-  render() {
-    // TODO: Complete layer for board
-    // TODO: Complete layer for pieces
+const Board = (props: Props) => {
+  let width = 600;
+  let height = 600;
+
+  props = {
+    ...props,
+    ...testProps
+  };
+
+  // TODO: Complete layer for board
+  let boardImage = props.gameSpec.board.downloadURL;
+  let boardLayer = (
+    <CanvasImage height={height} width={width} src={boardImage} />
+  );
+
+  // TODO: Complete layer for pieces
+  let piecesLayer = props.matchInfo.matchState.map((piece, i) => {
+    let pieceSpec = props.gameSpec.pieces[piece.currentImageIndex];
     return (
-      <Stage>
-        <Layer />
-        <Layer>
-          {this.props.pieces.length
-            ? this.props.pieces.map((piece: Piece, key: number) => {
-                return <div key={key}>{piece}</div>;
-              })
-            : ''};
-        </Layer>
-      </Stage>
+      <BoardPiece
+        key={'piece' + i}
+        height={pieceSpec.element.height * height / props.gameSpec.board.height}
+        width={pieceSpec.element.width * width / props.gameSpec.board.height}
+        x={piece.x * width / 100}
+        y={piece.y * height / 100}
+        src={'placeholder'}
+        element={pieceSpec.element}
+      />
     );
-  }
-}
+  });
+
+  return (
+    <>
+      <AppBar
```

Comment:
AppBar is usually a top level element that goes into a top level container. I am not sure if this is the right place to keep it. Maybe move it `App` for navigation purposes and have a using store to get the current selected game?